libs/store/src/lib/domain/query/query.ts

```diff
@@ -0,0 +1,243 @@
+import {
+    Type
+} from '@angular/core';
+import {
+    Observable,
+    of
+} from 'rxjs';
+import {
+    EntityCollectionService,
+    QueryParams
+} from '@ngrx/data';
+
+import { Predicate } from './grammer/predicate';
+import { QueryAdapter } from './query-adapter';
+
+export interface OrderBy <TModel, TProperty extends keyof TModel> {
+    field: TProperty;
+    order?: 'ASCENDING' | 'DESCENDING';
+}
+
+/**
+ * @todo We may need a method for end-users to add custom query parameters.
+ */
+export class Query<TModel> {
+
+    /** @hidden - stores current keyword */
+    _keyword: string;
+
+    /** @hidden - stores current predicate */
+    _predicate: Predicate<TModel>;
+
+    /** @hidden - stores current page size */
+    _pageSize: number;
+
+    /** @hidden - stores current index offset */
+    _offset: number;
```

Comment:
What if we name these internal properties to match what you ahve in tests and what is also in Ariba REST API? 

top, skip

Reply:
I can rename to "top" and "skip".